crates/db_schema/src/impls/tag.rs

```diff
@@ -13,23 +19,92 @@ use diesel::{
   ExpressionMethods,
   QueryDsl,
 };
-use diesel_async::RunQueryDsl;
-use lemmy_db_schema_file::schema::tag;
+use diesel_async::{scoped_futures::ScopedFutureExt, RunQueryDsl};
+use lemmy_db_schema_file::schema::{post_tag, tag};
 use lemmy_utils::error::{LemmyErrorExt, LemmyErrorType, LemmyResult};
+use std::collections::HashSet;
 
 impl Tag {
-  pub async fn get_by_community(
+  pub async fn read_for_community(
     pool: &mut DbPool<'_>,
-    search_community_id: CommunityId,
+    community_id: CommunityId,
   ) -> LemmyResult<Vec<Self>> {
     let conn = &mut get_conn(pool).await?;
     tag::table
-      .filter(tag::community_id.eq(search_community_id))
+      .filter(tag::community_id.eq(community_id))
       .filter(tag::deleted.eq(false))
       .load::<Self>(conn)
       .await
       .with_lemmy_type(LemmyErrorType::NotFound)
   }
+
+  pub async fn update_many(
+    pool: &mut DbPool<'_>,
+    community: &Community,
+    mut forms: Vec<TagInsertForm>,
+  ) -> LemmyResult<()> {
+    let known_tags = Tag::read_for_community(pool, community.id).await?;
```

Comment:
Move this read up a level and pass it in. You are doing validation here when it should be done before this. 

Getting the intersection / trimmed down subset of allowed tags needs to be a common function, done before any inserts.

Reply:
Bump on this one, make sure the `known_tags` is passed into this function.